Marlin/src/gcode/bedlevel/G35.cpp

```diff
@@ -178,11 +158,11 @@ void GcodeSuite::G35() {
   // the probe deployed if it was successful.
   probe.stow();
 
+  // Move XYZ to wait position
+  move_to_tramming_wait_pos();
+
   // After this operation the Z position needs correction
   set_axis_never_homed(Z_AXIS);
-
-  // Home Z after the alignment procedure
-  process_subcommands_now_P(PSTR("G28Z"));
 }
```

Comment:
I think we should keep homing after the wizard. @qwewer0 seems you dont like homing 😄. We may could do it like as its handled in the probe offset wizard:
```
    // If probe is used for homing and there is no fixed endstop, rehome Z
    #if EITHER(Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN, USE_PROBE_FOR_Z_HOMING)
      set_axis_never_homed(Z_AXIS);
      queue.inject_P(PSTR("G28Z"));
    #endif

  // Move XYZ to wait position
  move_to_tramming_wait_pos();
```

Reply:
My idea to why not to home after the G35 is, because after the bed adjustment we would home the printer either way when printing a gcode file, or doing another G35 which home too at the start, so homing twice would be a waste of time.

Reply:
But I'm still open in the idea, just not convinced...

Reply:
> I would consider to make a change some lines above (beginning of the percedure) from home_all_axes(); to queue.inject_P(PSTR("G28O"));

I don't think it would be a good idea to change the first home to an optional one, because if the previous bed adjustment after the G35 is too great, then the `Z_PROBE_LOW_POINT` would trigger and give use an error.
But I might be wrong in that, but still...

Reply:
While inside of a G-code command no other G-code commands in the queue will execute. So you would never want to use `queue.inject_P` to do something that you want to happen before your currently-processing G-code exits. Instead, you should use `process_subcommands_now_P` for commands that you want be executed here and now.

Reply:
> 
> 
> > I would consider to make a change some lines above (beginning of the percedure) from home_all_axes(); to queue.inject_P(PSTR("G28O"));
> 
> With small changes on the screws the bed center could stay within the `Z_PROBE_LOW_POINT` area, but if the bed was badly tilted, it isn't guaranteed that the middle of the bed would stay within the `Z_PROBE_LOW_POINT` area. And we cannot be sure that homing position is at the center of the bed.
> So I don't think it would be a good idea to change the first home to an optional one, because if the previous bed adjustment after the G35 is too great, then the `Z_PROBE_LOW_POINT` would trigger and give use an error.
> Even if I'm wrong in this, I still think the first home should stay there...

I agree 👍 

And I stay with my opinion and suggest the following changes for the end of the tramming process:

```
  // If probe is used for homing and there is no fixed endstop, rehome Z
  #if HOMING_Z_WITH_PROBE
    set_axis_never_homed(Z_AXIS);
    process_subcommands_now_P(PSTR("G28Z"));
  #endif

  // Move XYZ to wait position
  move_to_tramming_wait_pos();
```